src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -139,12 +141,144 @@ class IndexLogEntryTest extends SparkFunSuite {
+    withTempPath { p =>
+      // Prepare some files and directories. This time, we make multi-level directory tree.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+      val d1 = Files.createTempDirectory(dir, "d1")
+      val d1Path = PathUtils.makeAbsolute(d1.toString)
+      Files.createTempFile(d1, "f4", "")
+
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val content = Content.fromDirectory(dirPath)
+
+      val expected =
+        (fs.listStatus(dirPath).filter(_.isFile) ++ fs.listStatus(d1Path)).map(_.getPath).toSet
+
+      val actual = content.files.toSet
+      assert(actual.equals(expected))
+    }
+  }
+
+  test("Test fromPath api creates the correct Content object.") {
+    withTempPath { p =>
+      // Prepare some files and directories.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+
+      // Create expected Content object.
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val fileInfos = fs.listStatus(dirPath).map(FileInfo(_))
+      val leafDir = Directory(dirPath.getName, fileInfos)
+      val expected = {
+        val rooDirectory = TestUtils.splitPath(dirPath.getParent).foldLeft(leafDir) {
+          (accum, name) =>
+            Directory(name, Seq(), Seq(accum))
+        }
+
+        Content(rooDirectory, NoOpFingerprint())
+      }
+
+      // Create actual Content object.
+      val actual = Content.fromDirectory(dirPath)
+
+      // Compare.
+      assert(actual.equals(expected))
+    }
+  }
+
+  test("Test fromLeafFiles api creates the correct Content object.") {
+    withTempPath { p =>
+      // Prepare some files and directories.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+
+      // Create expected Content object.
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val fileStatuses = fs.listStatus(dirPath)
+      val fileInfos = fileStatuses.map(FileInfo(_))
+      val leafDir = Directory(dirPath.getName, fileInfos)
+
+      val expected = {
+        val rooDirectory = TestUtils.splitPath(dirPath.getParent).foldLeft(leafDir) {
+          (accum, name) =>
+            Directory(name, Seq(), Seq(accum))
```

Comment:
`expected` has no files? The `actual` should have some files no? To me, it's much easier to understand if you handcraft the expected since this depends on `splitPath` and `foldLeft`.

Reply:
The first parameter in foldLeft (`leafDir`) has the list of files. Let me add the handcrafting logic.